src/main/java/com/google/api/generator/engine/ast/Variable.java

```diff
@@ -47,6 +47,7 @@ public Variable build() {
 
       Variable variable = autoBuild();
       Preconditions.checkState(!variable.type().typeKind().equals(TypeNode.TypeKind.VOID));
+      Preconditions.checkState(!variable.type().equals(TypeNode.NULL));
```

Comment:
Consider adding an error message, e,g. `checkState(condition, "Variable type cannot be null")`. Not part of this PR, but the VOID one above should have one as well.

Reply:
Let me add in message in this PR.